torch/backends/xeon/_cpu_info.py

```diff
@@ -0,0 +1,455 @@
+    - [int] CPU index
+    - [int] Core index
+    - [int] Numa node index
+    - [int] Socket index
+    - [bool] is a physical core or not
+    - [float] maxmhz
+    - [bool] is a performance core
+    """
+
+    def __init__(self, lscpu_txt="", headers=None) -> None:
+        if headers is None:
+            headers = {}
+        self.cpu = -1
+        self.core = -1
+        self.socket = -1
+        self.node = -1
+        self.is_physical_core = True
+        self.maxmhz = 0.0
+        self.is_p_core = True
+        if lscpu_txt != "" and len(headers) > 0:
+            self.parse_raw(lscpu_txt, headers)
+
+    def parse_raw(self, cols, headers):
+        self.cpu = int(cols[headers["cpu"]])
+        self.core = int(cols[headers["core"]])
+        if "node" in headers:
+            self.node = int(cols[headers["node"]])
+            self.socket = int(cols[headers["socket"]])
+        else:
+            self.node = int(cols[headers["socket"]])
+            self.socket = int(cols[headers["socket"]])
+        if "maxmhz" in headers:
+            self.maxmhz = float(cols[headers["maxmhz"]])
+
+    def __str__(self):
+        return f"{self.cpu}\t{self.core}\t{self.socket}\t{self.node}\t{self.is_physical_core}\t{self.maxmhz}\t{self.is_p_core}"
+
+
+class CPUPool(list):
+    """
+    List of CoreInfo objects
+    """
+
+    def __init__(self) -> None:
+        super().__init__(self)
+
+    def get_ranges(self, l):
+        for a, b in itertools.groupby(enumerate(l), lambda pair: pair[1] - pair[0]):
+            bl = list(b)
+            yield bl[0][1], bl[-1][1]
+
+    def get_pool_txt(self, return_mode="auto"):
+        cpu_ids = [c.cpu for c in self]
+        cpu_ranges = list(self.get_ranges(cpu_ids))
+        cpu_ids_txt = ",".join([str(c) for c in cpu_ids])
+        cpu_ranges_txt = ",".join([f"{r[0]}-{r[1]}" for r in cpu_ranges])
+        node_ids_txt = ",".join(
+            [
+                str(n)
+                for n in sorted(list(set([c.node for c in self])))  # noqa: C414,C403
+            ]
+        )
+        ret = {"cores": "", "nodes": node_ids_txt}
+        if return_mode.lower() == "list":
+            ret["cores"] = cpu_ids_txt
+        elif return_mode.lower() == "range":
+            ret["cores"] = cpu_ranges_txt
+        else:
+            if len(cpu_ids) <= len(cpu_ranges):
+                ret["cores"] = cpu_ids_txt
+            else:
+                ret["cores"] = cpu_ranges_txt
+        return ret
+
+
+class CPUPoolList:
+    """
+    Get a CPU pool with all available CPUs and CPU pools filtered with designated criterias.
+    """
+
+    def __init__(self, logger=None, lscpu_txt="") -> None:
+        self.pool_all = CPUPool()
+        self.pools_ondemand: list[CPUPool] = []
+
+        self.logger = logger
+        if platform.system() == "Windows":
+            raise RuntimeError("Windows platform is not supported!!!")
+        elif platform.system() == "Linux":
```

Comment:
IMO this should be `if platform.system() == "Linux" and platform.machine() == "x86_64": ... else: raise RuntimeError(f"Unsupported platform {platform.system()}")` to cover both Windows, MacOS as well as LinuxARM